### PR TITLE
Handle crashes better.

### PR DESCRIPTION
**Pull request by GauntletWizard**
Complete rewrite to duplicate less, and handle container failure better. Exit strategy should use last-autosave if last-autosave is newer than exit savefile (Which indicates that the server shutdown uncleanly). Otherwise, it should behave identically, which a bunch less duplication.

I started off with some less-invasive fixes, if you'd prefer :)

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,62 +1,40 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (luaduck) [inline entrypoint.sh]: Is bash necessary here? We're adding another dependency here if we ever want to slim down the image even more.

File: k8s/openttd.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  template:
+    metadata:
+      labels:
+        app: openttd
+      name: openttd
+    spec:
+      containers:
+      - args: []
+        env: []
+        image: gauntletwizard/misc:openttd
```
Comment (luaduck) [inline k8s/openttd.yaml]: Wrong image name, please change to redditopenttd/openttd:latest

File: k8s/openttd.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        ports:
+        - containerPort: 3979
+          name: openttd
+        volumeMounts:
+        - mountPath: /config
+          name: openttd-data
+      securityContext:
+        runAsGroup: 911
+        runAsUser: 911
+      serviceAccountName: openttd
```
Comment (luaduck) [inline k8s/openttd.yaml]: Does this work if the serviceAccount isn't defined anywhere?

File: k8s/openttd.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  updateStrategy:
+    type: RollingUpdate
+  volumeClaimTemplates:
+  - metadata:
```
Comment (luaduck) [inline k8s/openttd.yaml]: Would this be tidier as a reference to a PVC?

**Comment by duckfullstop** [comment]
Closing as stale (we now include a Kustomize configuration if you're interested)

**Pull request closed without merging**